src/data_collection.py

```diff
@@ -269,6 +269,98 @@ def print_individual_in_table(file_name):
                 dictionary[key]["FORMAT"],
             ]
         )
+
+    print(data_table)
+
+
+def get_testing_commit_info(json_file_name):
+    """Get repo info about commits to or not to testing."""
+
+    current_data = json_handler.get_dict_from_json_file(json_file_name)
+    # creates a hashmap where the key is the authors username
+    data_dict = {}
+    # Check if RAW_DATA is in json
+    if "RAW_DATA" in current_data.keys():
+        for key in current_data["RAW_DATA"]:
+            author = key["author_name"]
+            filepaths = key["filepath"]  # get filepaths of file modified
+
+            # NOTE check date compatibility with json
+            # date = "N/A"
+            # check if the key already in in the dicitionary
+            if author in data_dict:
+                # condition passed, adds one to the number of commits
+                data_dict[author]["COMMITS"] += 1
+            else:
+                # condition fails, creates a new key and adds empty data
+                data_dict[author] = {
+                    "COMMITS": 1,
+                    "COMMITS_TO_TESTING": 0,
+                    "COMMITS_ELSEWHERE": 0,
+                    "PERCENT_TO_TESTING": 0,
+                    "PERCENT_NOT_TO_TESTING": 0,
+                }
+
+            count = 0  # when 0 the current commit has no changes to tests
+            for filepath in filepaths:
+                if (
+                    count is 0
+                ):  # when 0, current commit no testing changes; if not 0, this commit already had testing changes
+                    if filepath:
+                        if "test" in filepath:
+                            data_dict[author][
+                                "COMMITS_TO_TESTING"
+                            ] += 1  # the current commit has a change to testing
+                            count = 1  # found a change to testing for this commit
+                        else:
+                            pass
+                else:
+                    pass
+            # Sort list to ensure consistency when testing
+        # iterate through the data to do final calculations
+        for key in data_dict:
+            data_dict[key]["PERCENT_TO_TESTING"] = (
+                data_dict[key]["COMMITS_TO_TESTING"] / data_dict[key]["COMMITS"]
+            ) * 100
+
+            data_dict[key]["COMMITS_ELSEWHERE"] = (
+                data_dict[key]["COMMITS"] - data_dict[key]["COMMITS_TO_TESTING"]
+            )
+
+            data_dict[key]["PERCENT_NOT_TO_TESTING"] = (
+                data_dict[key]["COMMITS_ELSEWHERE"] / data_dict[key]["COMMITS"]
+            ) * 100
```

Comment:
We've discussed some of these calculations with the overall evaluations team, we thought that all calculations should be done in the evaluations file they are working on. Please discuss this with @bagashvilit  and @MaddyKapfhammer who are working on this.

Reply:
I am leaving this as it is right now @noorbuchi because I see things like `data_dict[key]["MODIFIED"]` and `data_dict[key]["TOTAL"]` being calculated in exactly the same fashion.

This can be refactored later by feature teams if necessary.